engineering_tools/carma

```diff
@@ -339,6 +339,174 @@ carma__stop() {
     fi
 }
 
+carma__swarm() {
+
+    if [ "$1" = "deploy" ]; then
+
+        echo "Checking Docker Swarm status..."
+        if [ "$(docker info | grep Swarm | sed 's/Swarm: //g')" == " inactive" ]; then
+            echo "Initalizing Docker Swarm..."
+            docker swarm init --advertise-addr 192.168.88.100 # consider making a variable
+        else
+            echo "Docker Swarm active"
+        fi
+
+        # if carma-config-data volume doesn't exist exit and request update
+        if [ "$(docker volume ls | grep carma-config-data | sed 's/local     //g')" ]; then
+            echo "Found carma-config-data volume"
+        else
+            echo "Missing carma-config-data volume, add with \"carma swarm update-config <image tag>\""
+            exit 1
+        fi
+
+        # carma-web-ui will always run as a background process regardless of deploy argument
+        docker run --rm -v carma-config-data:/opt/carma/vehicle/config --entrypoint sh busybox:latest -c \
+        'cat /opt/carma/vehicle/config/docker-compose-background-pc1.yml' | \
+        docker stack deploy --compose-file - carma-background
+
+
+        # Run part of the stack on a single host. Mostly for debugging purposes off of the dual PC setup
+        if [ "$2" = "single" ]; then 
+            echo "Deploying CARMA to single host Swarm stack..."
+
+            docker run --rm -v carma-config-data:/opt/carma/vehicle/config --entrypoint sh busybox:latest -c \
+            'cat /opt/carma/vehicle/config/docker-compose-pc1.yml' | \
+            docker stack deploy --compose-file - carma-pc1
+
+
+        # For use with two connected PCs with known IPs
+        # Saves the generated swarm join token and ssh's into the worker to execute and deploy
+        elif [ "$2" = "dual" ]; then
+            echo "Deploying CARMA to dual compute Swarm stack..."
+
+            local TKN=$(docker swarm join-token -q worker)
+            ssh dev@192.168.88.101 "docker swarm join --token $TKN 192.168.88.100:2377"
+
+            docker run --rm -v carma-config-data:/opt/carma/vehicle/config --entrypoint sh busybox:latest -c \
+            'cat /opt/carma/vehicle/config/docker-compose-pc1.yml' | \
+            docker stack deploy --compose-file - carma-pc1
+
+            docker run --rm -v carma-config-data:/opt/carma/vehicle/config --entrypoint sh busybox:latest -c \
+            'cat /opt/carma/vehicle/config/docker-compose-pc2.yml' | \
+            docker stack deploy --compose-file - carma-pc2
+            
+        
+        elif [ ! -z "$2" ]; then
+            echo "Unrecognized argument \"swarm deploy $2\""
+        fi
+
+
+    # Updates the carma-config-data volume based on a input tag string given by the user
+    elif [ "$1" = "update-config" ]; then
```

Comment:
Not the end of the world, but I think it might be worth considering 'set' for this verb, as that's the same paradigm as the non-swarm usage. Could be less confusing for users, `carma config set` and `carma swarm config set` or something like that?

Breaking the pattern is fine if we're intentional about it though, if this is doing something fundamentally different from `set` in your mind.